### PR TITLE
Transition to JS

### PR DESCRIPTION
**Pull request by alikhamesy**
5 years late to the party, but I transitioned to js as stated in #16. It's pretty much a 1:1 mapping from the old code, except a few ES6/7 features. Also for the build step I put in minified for babel so index would be a little smaller but I'm not sure if that matters.

### CODE REVIEW

**Comment by sergeyt** [comment]
@alikhamesy wow 😄. Thanks for the PR. One small suggestion - lets format JS with Prettier

**Comment by alikhamesy** [comment]
> lets format JS with Prettier
Sounds good, I'll set that up!

**Comment by alikhamesy** [comment]
@sergeyt question about `file.index`, should it `['sha...sha', 'fileMode']` or just `'sha...sha'`? right now its the former but I get the feeling it should be the latter?

**Comment by sergeyt** [comment]
> @sergeyt question about `file.index`, should it `['sha...sha', 'fileMode']` or just `'sha...sha'`? right now its the former but I get the feeling it should be the latter?

I am not sure. I suggest keeping it as it is for now. Maybe need to raise separate discussion for this change

**Review by sergeyt — CHANGES_REQUESTED** [review]

File: .eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  "extends": ["eslint:recommended", "plugin:prettier/recommended"],
+  "plugins": ["prettier"],
+  "parserOptions": {
+    "ecmaVersion": 12
+  },
+  "rules": {
+    "prettier/prettier": [
+      "error",
+      {
+        "tabWidth": 2,
```
[the remark this answers is not in the document]
Comment (alikhamesy) [reply .eslintrc.json]: done :)

**Review by sergeyt — APPROVED** [review]

**Comment by sergeyt** [comment]
@alikhamesy nice, thank you

**Pull request merged**